src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/SymbolTable.cs

```diff
@@ -747,7 +748,7 @@ private static List<object> BuildDeclarationChain(Type callingType)
                 {
                     MethodBase methodBase = t.DeclaringMethod;
                     bool bAdded = false;
-                    foreach (MethodInfo methinfo in Enumerable.Where(t.DeclaringType.GetRuntimeMethods(), m => m.HasSameMetadataDefinitionAs(methodBase)))
+                    foreach (MethodInfo methinfo in Enumerable.Where(t.DeclaringType.GetMethods(EverythingBindingFlags), m => m.HasSameMetadataDefinitionAs(methodBase)))
```

Comment:
Also get rid of the use of Linq by moving the condition inside the for-loop? It will make the code smaller and faster overall.

Reply:
Yup, thanks.